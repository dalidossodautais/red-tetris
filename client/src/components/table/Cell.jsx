import styled from "styled-components";

export default styled.div.attrs((props) => {
  const added = {
    color: props.theme.primaryTextColor,
  };
  return added;
})`
  align-items: center;
  color: ${(props) => props.color};
  display: flex;
  font-size: 15px;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: ${(props) => props.fontWeight};
  height: 35px;
  line-height: 35px;
  margin: 10px;
  text-align: center;
  width: ${(props) => props.width};
`;
