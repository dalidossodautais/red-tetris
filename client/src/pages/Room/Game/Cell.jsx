import React from "react";
import styled from "styled-components";

function darken(colorPart) {
  return (colorPart * 3) / 4;
}
function lighten(colorPart) {
  return (colorPart * 5) / 4;
}

const colors = (() => {
  const basics = {
    I /* Cyan */: "rgb(0,184,212)",
    J /* Pink */: "rgb(233,30,99)",
    L /* Orange */: "rgb(255,109,0)",
    O /* Yellow */: "rgb(255,214,0)",
    S /* Teal */: "rgb(0,150,136)",
    T /* Deep Purtle */: "rgb(103,58,183)",
    Z /* Red */: "rgb(244,67,54)",
    full: "rgb(64,64,64)",
    shadow: "rgb(192,192,192)",
    null: "rgb(220,220,220)",
  };
  const merged = {};
  Object.entries(basics).forEach(([key, basic]) => {
    const parted = basic.match(/\d+/g);
    merged[key] = {
      basic,
      dark: `rgb(${darken(parted[0])},${darken(parted[1])},${darken(
        parted[2]
      )})`,
      light: `rgb(${lighten(parted[0])},${lighten(parted[1])},${lighten(
        parted[2]
      )})`,
    };
  });
  return merged;
})();

const StyledCell = styled.div.attrs((props) => {
  if (!props.transparent)
    return {
      backgroundColor: props.basic,
      borderBottomColor: props.dark,
      borderLeftColor: props.light,
      borderRightColor: props.dark,
      borderTopColor: props.light,
    };
  return {
    backgroundColor: "transparent",
    borderBottomColor: "transparent",
    borderLeftColor: "transparent",
    borderRightColor: "transparent",
    borderTopColor: "transparent",
  };
})`
  background-color: ${(props) => props.backgroundColor};
  border-bottom-color: ${(props) => props.borderBottomColor};
  border-left-color: ${(props) => props.borderLeftColor};
  border-right-color: ${(props) => props.borderRightColor};
  border-style: solid;
  border-width: ${(props) => props.size * 0.05}px;
  border-top-color: ${(props) => props.borderTopColor};
  height: ${(props) => props.size * 0.9}px;
  opacity: ${(props) => props.opacity};
  width: ${(props) => props.size * 0.9}px;
`;

export default function Cell({ cell = null, empty, opacity, size }) {
  const { basic, dark, light } = colors[cell];
  return (
    <StyledCell
      basic={basic}
      dark={dark}
      light={light}
      opacity={opacity}
      size={size}
      transparent={empty && !cell}
    />
  );
}
