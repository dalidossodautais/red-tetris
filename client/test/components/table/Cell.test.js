import React from "react";
import renderer from "react-test-renderer";
import Cell from "../../../src/components/table/Cell";

describe("Cell", () => {
  test("Standard", () => {
    const component = renderer.create(<Cell />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
