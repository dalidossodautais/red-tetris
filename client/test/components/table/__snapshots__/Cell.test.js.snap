// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Standard 1`] = `
<div
  className="sc-bdfBwQ jbGekO"
/>
`;

exports[`Cell Standard 2`] = `
<div
  className="sc-bdfBwQ jbGekO"
/>
`;
